nameserver/server.go

```diff
@@ -32,7 +20,94 @@ func makeDNSFailResponse(r *dns.Msg) *dns.Msg {
 	return m
 }
 
-func queryHandler(lookups []Lookup) dns.HandlerFunc {
+type DNSServer struct {
+	config *dns.ClientConfig
+	zone   Zone
+	iface  *net.Interface
+	port   int
+	udpSrv *dns.Server
+	tcpSrv *dns.Server
+}
+
+// Creates a new DNS server with a given config
+func NewDNSServerWithConfig(config *dns.ClientConfig, zone Zone, iface *net.Interface, port int) (*DNSServer, error) {
+	return &DNSServer{
+		config: config,
+		zone:   zone,
+		iface:  iface,
+		port:   port,
+	}, nil
+}
+
+// Creates a new DNS server
+func NewDNSServer(zone Zone, iface *net.Interface, port int) (*DNSServer, error) {
+	config, err := dns.ClientConfigFromFile("/etc/resolv.conf")
+	if err != nil {
+		return nil, err
+	}
+	return NewDNSServerWithConfig(config, zone, iface, port)
+}
+
+// Start the DNS server
+func (s *DNSServer) Start() error {
```

Comment:
This function never actually appears to return an error. Instead every error is fatal. Which is horrific. (yes, I know some of this is a remnant of the previous code).